ouroboros-consensus/docs/ReleaseProcess.md

```diff
@@ -0,0 +1,140 @@
+- Whenever we're also maintaining some lesser MAJOR.MINOR versions of a package, then some work for that package will also be happening on the respective _release branches_.
+
+A _release branch_ is a branch dedicated to the maintenance of some older release of a package.
+
+RULE: If we have released a package FOO-X.Y.Z, then the release-FOO-X.Y.x branch MUST exist and its first-parent history MUST include the commit released as FOO-X.Y.Z.
+
+There are two kinds of work on release branches.
+Most of the time, that work is immitating some work that was done on the main branch.
+EG we fixed a bug on the main branch and we're backporting that fix onto a release branch (hopefully it's a trivial cherry-pick).
+Rarely, though, there will be fresh work done on a release branch.
+EG Consider the following possible timeline.
+
+- We release FOO-2.4.1 on Monday, and FOO-2.4.1 completely reworked some functionality.
+- We then realize on Tuesday that there was a severe bug in the old functionality.
+- We fix the bug by merging a PR that targets the FOO-2.3.x release branch (since the bug no longer exists on the main branch!)
+- We release FOO-2.3.7 on Wednesday.
+
+*Remark*.
+Not every first-parent commit in the history of the release branch is a release commit.
+We will be merging multiple PRs into the release branch in order to prepare the next release from it, so some commits will just be the intermediates between two releases.
+
+*Remark*.
+The release branch for the greatest MAJOR.MINOR version of a package is somewhat degenerate.
+It's either equal to or a prefix of the main branch.
+The only time it MUST be updated to the tip of the main branch is when we cut a release of the package from the main branch, to satisfy the rule above about all X.Y.Z release commits being on the release-X.Y.x branch.
+
+## Rules for PRs
+
+We classify each PR as either a _fresh_ PR or as a _backport_ PR.
+
+- A fresh PR is doing work that has never been done before (new feature, new bugfix, etc).
+  Except in the rare circumstances discussed in the previous section, every fresh PR will target the main branch.
+- The primary objective of a backport PR is merely to immitate some fresh PR.
+
+The rules are then as follows.
+
+- RULE: A fresh PR MUST target the main branch, if possible.
+- RULE: A PR MUST NOT be merged into a branch unless it should be included in the next release from that branch.
+- We maintain a changelog for each package (one for FOO and one for BAR).
+- RULE: A PR MUST add a pending changelog entry for each package that it alters.
+- RULE: Each pending changelog entry MUST at the very least classify the alteration to that package as a _patch_ level change (eg a bugfix), a _minor_ change (eg a non-breaking interface change), or a _major_ change (eg a breaking interface change).
+
+*Remark*.
+Notice that we do not allow merging a PR that should not be included the subsequent release.
+This means some PR may be "done" but still should not be merged.
```

Comment:
So according to this rule, if we are not sure we will be releasing a new feature, eg UTxO-HD, the feature has to live in a branch, right?

Reply:
Yes. Please confirm: that's what we decided on in our call, right?

Reply:
Hmm. If, _unlike UTxO HD_, we do have an easy way to "include the feature but definitely hidden behind a flag", then we could go ahead and merge that. But we'd only merge it if we are _intending_ for it to be in the next release: the "experimental flag" will be part of the next release. (That's the point of the flag.)

Reply:
> Yes. Please confirm: that's what we decided on in our call, right?

I think so, but I'll ask the rest.